source/iofs/iofs.go

```diff
@@ -0,0 +1,36 @@
+// +build go1.16
+
+package iofs
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"net/http"
+
+	"github.com/golang-migrate/migrate/v4/source"
+	"github.com/golang-migrate/migrate/v4/source/httpfs"
+)
+
+func init() {
+	source.Register("iofs", &Iofs{})
+}
+
+// Iofs is a source driver for io/fs#FS.
+type Iofs struct {
+	httpfs.PartialDriver
+}
+
+// Open by url does not supported with Iofs.
+func (i *Iofs) Open(url string) (source.Driver, error) {
+	return nil, errors.New("iofs driver does not support open by url")
+}
+
+// WithInstance wraps io/fs#FS as source.Driver.
+func WithInstance(fsys fs.FS, path string) (source.Driver, error) {
+	var i Iofs
+	if err := i.Init(http.FS(fsys), path); err != nil {
```

Comment:
Converting from `fs.FS` to `http.FileSystem` may fail later if the underlying struct doesn't implement `io.Seeker`. `Seek()` isn't used in the `httpfs` source driver right now, but if that changes, this would break.

See: https://tip.golang.org/src/net/http/fs.go
```golang
func (f ioFile) Seek(offset int64, whence int) (int64, error) {
	s, ok := f.file.(io.Seeker)
	if !ok {
		return 0, errMissingSeek
	}
	return s.Seek(offset, whence)
}
```